### ISSUE TITLE
Jupyter not reloading data with xarray

### ISSUE THREAD

**Issue by lukaskroenert** [jupyter/notebook#5101]
I load a Netcdf4 file with xarray and can see the correct content.

Cell 1
```python3
import xarray as xr
data_path = 'path_to_file.nc'
```

Cell 2
```python3
data = ''
data = xr.open_dataset(data_path)
print(data)
```
 Afterward I change, or even delete this file on the disk. Now if I run again the the cell 2 again, the 'old' file is still used for the output.

This looks like the data is somehow saved in a cache, even though I use `data=''`.

This seams not to happen in ipython and jupyer-lab and jupyter-lab. Still, it could be an issue of xarray.

**Comment by jasongrout** [comment]
Did you mean to do `data_path=''` in that second cell?

**Comment by lukaskroe** [comment]
Yes, I do this to explicitly clear the variable. However, although I removed the data from the disk and I cleared the variable with `data ='' `, it is still saved somewhere. 

I expect to get an error message after clearing the data from the disk since no data should loaded. Instead I get the previously opened data.

**Comment by jasongrout** [comment]
You clear `data`, but use `data_path`

**Comment by lukaskroe** [comment]
Yes, that is the point. After the first run of the cell, I delete the data with `rm data_path` in some terminal so that at `data_path` no file exists anymore on the disk.
So I get for 
```bash
ls: data_path: No such file or directory
```
Which is correct, since I deleted it before and I expect similar in jupyter. However, the previously loaded data is still loaded there instead of getting an error message.

**Comment by lukaskroe** [comment]
I get similar behaviour If I change the original nc file. The new version of the nc file is than not loaded and I still get the output of the old version.

**Comment by jasongrout** [comment]
Change `data=''` to `data_path=''` in your second cell

**Comment by jasongrout** [comment]
Ah, I see I totally misunderstood the issue. Sorry. Yeah, looks like an xarray issue or testing issue, not a Jupyter issue.

**Comment by lukaskroe** [comment]
Ah ok, thank you Jason, than I'll report this issue to xarray. The only point is that I'm not sure it seems to work fine in ipython.

**Comment by jasongrout** [comment]
If I were you, I would just make absolutely *sure* you were changing/deleting the right file in your tests.

**Comment by lukaskroe** [comment]
This, I definitely made sure :).
I also tried to give a more detailed example and reported it to 
https://github.com/pydata/xarray/issues/3593
